docs/registry.md

```diff
@@ -0,0 +1,341 @@
+# Registry
+
+The registry is a place where applications metadata are stored and versioned. It can be used by a cozy 
+
+We define an application registry as an API. This should allow us to defer the real implementation of the registry storage and allow different store implementations.
+
+## Objects
+
+Two type of objects are managed in the registry: applications and versions.
+
+### Application
+
+An application object contains the following fields:
+
+- `name`: the application name
+- `type`: the application type ("webapp" or "konnector")
+- `editor`: the application editor name
+- `category`: the application category
```

Comment:
I'll a modifying route for the app object which is mutable (contrary to the version objects). The description of an application is different from the version description which describes what the version changes.

For the tags... I think it is indeed better to put it in the application object. For the license, I am not sure but I'll move it to the version object.